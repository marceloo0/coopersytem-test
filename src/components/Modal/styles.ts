import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
`;
export const Content = styled.View`
  background-color: #fff;
  height: 240px;
`;
export const Title = styled.Text`
  font-size: 28px;
  font-weight: bold;
  margin-bottom: 20px;
`;
export const Description = styled.Text`
  color: #6c6c6c;
  font-size: 18px;
`;
export const ContainerInfo = styled.View`
  align-items: center;
  justify-content: center;
  height: 180px;
  padding: 20px;
`;
