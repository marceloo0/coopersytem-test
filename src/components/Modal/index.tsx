import React from 'react';
import ModalComponent from 'react-native-modal';

import Bottom from '../Bottom';

import {Container, Content, Title, Description, ContainerInfo} from './styles';

interface TextProps {
  title: string;
  description: string;
  btnText: string;
}

interface ModalProps {
  visible: boolean;
  handleModal: any;
  texts: TextProps;
}

const Modal: React.FC<ModalProps> = ({texts, visible, handleModal}) => {
  return (
    <Container testID="modal-test">
      <ModalComponent isVisible={visible}>
        <Content>
          <ContainerInfo>
            <Title>{texts.title}</Title>
            <Description>{texts.description}</Description>
          </ContainerInfo>
          <Bottom onPress={handleModal} text={texts.btnText} />
        </Content>
      </ModalComponent>
    </Container>
  );
};

export default Modal;
