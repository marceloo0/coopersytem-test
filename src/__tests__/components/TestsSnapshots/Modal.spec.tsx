import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../../../components/Modal';

jest.useFakeTimers();

test('renders correctly', () => {
  const modal = renderer
    .create(
      <Modal
        visible={true}
        handleModal={() => {}}
        texts={{
          title: 'RESGATE EFETUADO',
          description:
            'O valor solicitado estará disponível em sua conta em 5 dias uteis.',
          btnText: 'NOVO RESGATE',
        }}
      />,
    )
    .toJSON();
  expect(modal).toMatchSnapshot();
});
