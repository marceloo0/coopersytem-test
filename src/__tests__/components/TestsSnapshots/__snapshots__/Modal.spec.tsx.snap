// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="modal-test"
>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View
        forwardedRef={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
    </TouchableWithoutFeedback>
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "height": 240,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 180,
                "justifyContent": "center",
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 28,
                  "fontWeight": "bold",
                  "marginBottom": 20,
                },
              ]
            }
          >
            RESGATE EFETUADO
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#6c6c6c",
                  "fontSize": 18,
                },
              ]
            }
          >
            O valor solicitado estará disponível em sua conta em 5 dias uteis.
          </Text>
        </View>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffff00",
                "height": 60,
                "justifyContent": "center",
                "marginTop": 15,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "blue",
                  "fontSize": 21,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            NOVO RESGATE
          </Text>
        </TouchableHighlight>
      </View>
    </View>
  </Modal>
</View>
`;
